### PR TITLE
Task #138557 feat: Add layout and type option field on TJUcm type form

### PR DESCRIPTION
**Pull request by sanjivani1812**
RIght now, ucm not providing different layout for form views. So for overcome this issue we are introduce new componentlayout field.

### CODE REVIEW

File: administrator/models/forms/type.xml [hunk 1]
```diff
@@ -26,6 +26,8 @@
 		<field name="modified_by" type="modifiedby" default="0" label="COM_TJUCM_FORM_LBL_TYPE_MODIFIED_BY" description="COM_TJUCM_FORM_DESC_TYPE_MODIFIED_BY" hidden="true" hint="COM_TJUCM_FORM_LBL_TYPE_MODIFIED_BY" />
 		<field name="modified_date" type="timecreated" label="COM_TJUCM_FORM_LBL_TYPE_MODIFIED_DATE" description="COM_TJUCM_FORM_DESC_TYPE_MODIFIED_DATE" hidden="true" hint="COM_TJUCM_FORM_LBL_TYPE_MODIFIED_DATE" />
 		<field name="version_note" type="text" label="JGLOBAL_FIELD_VERSION_NOTE_LABEL" description="JGLOBAL_FIELD_VERSION_NOTE_DESC" class="inputbox" size="45" labelclass="control-label" />
+		<field name="layout" type="componentlayout" label="COM_TJUCM_FORM_LBL_TYPE_LAYOUT" description="COM_TJUCM_FORM_DESC_TYPE_LAYOUT" useglobal="true" extension="com_tjucm" view="itemform"/>
```
Comment (ankush-maherwal) [inline administrator/models/forms/type.xml]: @sanjivani1812 
Where is the field type "componentlayout" defined?

File: administrator/models/forms/type.xml [hunk 2]
```diff
@@ -26,6 +26,8 @@
...
+		<field name="type_options" type="textarea" cols="5" rows="5" label="COM_TJUCM_FORM_LBL_TYPE_OPTIONS" description="COM_TJUCM_FORM_DESC_TYPE_OPTIONS" hint="COM_TJUCM_FORM_LBL_TYPE_TYPE_DESCRIPTION" />
```
Comment (ankush-maherwal) [inline administrator/models/forms/type.xml]: @sanjivani1812 
what is this field used for?

File: site/views/itemform/view.html.php [hunk 3]
```diff
@@ -85,6 +85,10 @@ public function display($tpl = null)
 
 		$TypeData = $tjUcmModelType->getItem($typeId);
 
+		// Set Layout to type view
+		$layout = $TypeData->itemform_layout;
```
Comment (ankush-maherwal) [inline site/views/itemform/view.html.php]: @sanjivani1812 
What is we update the extension?

Will the created layouts be removed?

File: administrator/models/forms/type.xml (hunk 2 above)
Comment (sanjivani1812) [reply administrator/models/forms/type.xml]: Now many of type need some extra options for a third party extension for this type option field to add here.

File: languages/administrator/en-GB/en-GB.com_tjucm.ini [hunk 4]
```diff
@@ -138,3 +138,7 @@ COM_TJUCM_PERMISSION_TYPE_ITEM_EDIT="Edit Item"
 COM_TJUCM_PERMISSION_TYPE_ITEM_EDITSTATE="Edit Item State"
 COM_TJUCM_PERMISSION_TYPE_ITEM_EDITOWN="Edit Own Item"
 COM_TJUCM_PERMISSION_TYPE_ITEM_DELETE="Delete Item"
+COM_TJUCM_FORM_LBL_TYPE_LAYOUT="Layout"
+COM_TJUCM_FORM_DESC_TYPE_LAYOUT="Selected layout used for this type"
+COM_TJUCM_FORM_LBL_TYPE_OPTIONS="Options"
+COM_TJUCM_FORM_DESC_TYPE_OPTIONS="Options"
```
Comment (ankush-maherwal) [inline languages/administrator/en-GB/en-GB.com_tjucm.ini]: @sanjivani1812 
add description for the field

File: site/views/itemform/view.html.php (hunk 3 above)
Comment (sanjivani1812) [reply site/views/itemform/view.html.php]: After update extension newed layout will not remove

**Pull request merged**